validator_instance.go

```diff
@@ -189,26 +189,38 @@ func (v *Validate) RegisterAlias(alias, tags string) {
 
 // RegisterStructValidation registers a StructLevelFunc against a number of types.
 //
+// It returns error when type being passed is a pointer.
 // NOTE:
 // - this method is not thread-safe it is intended that these all be registered prior to any validation
-func (v *Validate) RegisterStructValidation(fn StructLevelFunc, types ...interface{}) {
-	v.RegisterStructValidationCtx(wrapStructLevelFunc(fn), types...)
+func (v *Validate) RegisterStructValidation(fn StructLevelFunc, types ...interface{}) error {
+	return v.RegisterStructValidationCtx(wrapStructLevelFunc(fn), types...)
 }
 
 // RegisterStructValidationCtx registers a StructLevelFuncCtx against a number of types and allows passing
 // of contextual validation information via context.Context.
 //
+// It returns error when type being passed is a pointer.
 // NOTE:
 // - this method is not thread-safe it is intended that these all be registered prior to any validation
-func (v *Validate) RegisterStructValidationCtx(fn StructLevelFuncCtx, types ...interface{}) {
+func (v *Validate) RegisterStructValidationCtx(fn StructLevelFuncCtx, types ...interface{}) error {
 
 	if v.structLevelFuncs == nil {
 		v.structLevelFuncs = make(map[reflect.Type]StructLevelFuncCtx)
 	}
 
+	for _, t := range types {
+		if reflect.ValueOf(t).Kind() == reflect.Ptr {
```

Comment:
Have you tried using [Elem](https://golang.org/pkg/reflect/#Value.Elem) or [Indirect](https://golang.org/pkg/reflect/#Indirect) to automatically downgrade instead of returning an error?

Reply:
Thanks @joeybloggs, pushed a commit using Indirect to downgrade it